src/net_processing.cpp

```diff
@@ -2777,11 +2792,32 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         std::vector<CAddress> vAddrOk;
         int64_t nNow = GetAdjustedTime();
         int64_t nSince = nNow - 10 * 60;
+
+        // Update/increment addr rate limiting bucket.
+        const auto current_time = GetTime<std::chrono::microseconds>();
+        if (peer->m_addr_token_bucket < MAX_ADDR_PROCESSING_TOKEN_BUCKET) {
+            // Don't increment bucket if it's already full
+            const auto time_diff = std::max(current_time - peer->m_addr_token_timestamp, 0us);
```

Comment:
This could result in a very large number if `peer->m_addr_token_timestamp` is negative. Maybe a check that `peer->m_addr_token_timestamp` is positive?

Reply:
That variable is initialized to the current time, so it can't be negative.

Reply:
I was under the impression this variable is under peer (attacker) control, so could take on any value.

Reply:
That's incorrect. It's just our local time when the m_addr_token_bucket variable was last updated.